nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/heartbeat/ClusterProtocolHeartbeatMonitor.java

```diff
@@ -91,12 +91,8 @@ public ClusterProtocolHeartbeatMonitor(final ClusterCoordinator clusterCoordinat
         this.clusterNodesPath = zkClientConfig.resolvePath("cluster/nodes");
 
         String hostname = properties.getProperty(NiFiProperties.CLUSTER_NODE_ADDRESS);
-        if (hostname == null) {
-            try {
-                hostname = InetAddress.getLocalHost().getHostName();
-            } catch (UnknownHostException e) {
-                throw new RuntimeException("Unable to determine local hostname and the '" + NiFiProperties.CLUSTER_NODE_ADDRESS + "' property is not set");
-            }
+        if (hostname == null || hostname.trim().isEmpty()) {
```

Comment:
There's another similar null check against port `if (port == null) {`. I think it should be addressed, too.
We may be able to take advantage of our existing StringUtils.isBlank().